dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,158 @@
+# How to setup Windows machine for WSL testing
+
+To run our integration tests on WSL instances, we need to first setup a Windows machine
+to allows us to launch the WSL instances. Not only that, we also need to setup OpenSSH
+server to allow us to run ssh commands on the Windows machine.
+
+This how to guide will explain all the necessary steps for the setup, from launching the image
+from Azure to testing if WSL is correctly working on the machine.
+
+## Launching the image on Azure
+
+When launching the Windows instance on Azure, there a few configurations that need to be performed,
+otherwise we will not be able to launch WSL instances.
+
+First of all, name the machine as `wsl-test`. Currently, this is the name we expect on the
+integration tests.
+
+After that, select the `Windows 11` image and mark the `Security Type` as *Standard*.
+![windows_secuirty_type](windows-security-type.png)
+
+Set an username and password for the VM. Please set the username as *ubuntu*. 
+We will use this account setting to manually login into the machine to run some commands there directly.
+![windows_secuirty_type](windows-ssh-setup.png)
+
+Finally, remember to also select the SSH inbound port for the machine
+![windows_secuirty_type](windows-setup-port.png)
+
+
+## Configure Bastion to login into the machine
+
+To login into the machine and access its GUI, you need to setup Bastion.
+After launching the machine, go for the *Connect with Bastion* option:
+![windows_secuirty_type](windows-bastion.png)
+![windows_secuirty_type](windows-deploy-bastion.png)
+
+After deploying Bastion, you just need to present your username and password to login into the
+machine.
+
+## Configuring Hyper-V
+
+After login into the machine through Bastion, you need to configure Hyper-V by following
+this [tutorial](https://learn.microsoft.com/en-us/azure/lab-services/how-to-enable-nested-virtualization-template-vm-using-script?tabs=powershell)
+
+## Installing winget
+
+For each Ubuntu distro, we will get the installers directly from the Microsoft store.
```

Comment:
I'm not 100% sure I understand what you mean by this. Do you mean each Ubuntu release has an installer in the Microsoft store?

Reply:
Yes, exactly. I will make this clearer in the guide